plugins/woocommerce/src/Admin/API/ShippingPartnerSuggestions.php

```diff
@@ -0,0 +1,212 @@
+	}
+
+	/**
+	 * Return suggested shipping partners.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return \WP_Error|\WP_HTTP_Response|\WP_REST_Response
+	 */
+	public function get_suggestions( $request ) {
+		$should_display = $this->should_display();
+		$force_default  = $request->get_param( 'force_default_suggestions' );
+
+		if ( $should_display ) {
+			return Suggestions::get_suggestions();
+		} elseif ( false === $should_display && true === $force_default ) {
+			return rest_ensure_response( Suggestions::get_suggestions( DefaultShippingPartners::get_all() ) );
+		}
+
+		return rest_ensure_response( Suggestions::get_suggestions( DefaultShippingPartners::get_all() ) );
+	}
+
+	/**
+	 * Get the schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public static function get_suggestions_schema() {
+		$feature_def = array(
+			'type'  => 'array',
+			'items' => array(
+				'type'       => 'object',
+				'properties' => array(
+					'icon'        => array(
+						'type' => 'string',
+					),
+					'title'       => array(
+						'type' => 'string',
+					),
+					'description' => array(
+						'type' => 'string',
+					),
+				),
+			),
+		);
+		$layout_def  = array(
+			'type'       => 'object',
+			'properties' => array(
+				'image'    => array(
+					'type'        => 'string',
+					'description' => '',
+				),
+				'features' => $feature_def,
+			),
+		);
+
+		$item_schema = array(
+			'type'       => 'object',
+			'required'   => array( 'name', 'is_visible', 'available_layouts' ),
+			// require layout_row or layout_column. One of them must exist.
+			'anyOf'      => array(
+				array(
+					'required' => 'layout_row',
+				),
+				array(
+					'required' => 'layout_column',
+				),
+			),
+			'properties' => array(
+				'name'              => array(
+					'description' => __( 'Plugin name.', 'woocommerce' ),
+					'type'        => 'string',
+					'required'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'slug'              => array(
+					'description' => __( 'Plugin slug used in https://wordpress.org/plugins/{slug}.', 'woocommerce' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'layout_row'        => $layout_def,
+				'layout_column'     => $layout_def,
+				'description'       => array(
+					'description' => __( 'Description', 'woocommerce' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'learn_more_link'   => array(
+					'description' => __( 'Learn more link .', 'woocommerce' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'is_visible'        => array(
+					'description' => __( 'Suggestion visibility.', 'woocommerce' ),
+					'type'        => 'boolean',
```

Comment:
@moon0326 I just noticed that `is_visible` is not a boolean value in the API response. Should we change it to boolean?

![Screenshot 2023-03-14 at 09 53 21](https://user-images.githubusercontent.com/4344253/224871871-7387245b-e32f-4a3a-8325-6e46163837c6.png)

Reply:
Good catch!

It should be set to boolean after rule evaluation. Updated in https://github.com/woocommerce/woocommerce/pull/37155/commits/88bf0713fd7c2a054592f1d02292204aa5384c7d